TestProject/ClientApp/src/components/Layout.js

```diff
@@ -0,0 +1,40 @@
+﻿import React, { useEffect, useState } from 'react';
+import { Route } from 'react-router';
+import Main from "./Main"
+import UserProfile from "./UserProfile"
+import Header from "./Header";
+
+function Layout() {
+    const [userProfile, setUserProfile] = useState({
+        firstName: undefined, lastName: undefined, email: undefined, raiting: undefined
+    });
+    const [userLoggedIn, setUserLoggedIn] = useState(false)
+
+    useEffect(() => {
+        fetch("/user/getUserProfile", { mode: 'no-cors' })
+            .then(function (response) {
```

Comment:
```suggestion
            .then((response) => {
```
using lambda is more elegant